src/docs/grains/grain_persistence/relational_storage.md

```diff
@@ -93,122 +93,121 @@ public class AdoNetGrainStorageOptions
 
 ### 7. Make the design shardable
 
-When it makes sense, for instance in relational storage provider, make the design readily shardable. This means for instance no database dependent
-data (e.g. `IDENTITY`) and basically it means the information that distinguishes row data should build on only data from the actual parameters.
+When it makes sense, for instance in a relational storage provider, make the design readily shardable. For instance, this means using no database-dependent
+data (e.g. `IDENTITY`). Information that distinguishes row data should build on only data from the actual parameters.
 
 ### 8. Make the design easy to test
 
-Creating a new backend should be ideally as easy as translating one of the deployment scripts and adding a new connection string to tests assuming default
-parameters, check if a given database is installed and then run the tests against it.
+Creating a new backend should ideally be as easy as translating one of the deployment scripts, adding a new connection string to tests (assuming default
+parameters), check if a given database is installed, and then run the tests against it.
 
-### 9. Taking into account the previous points, make both porting scripts for new backends and modifying already deployed backend scripts as transparent as possible
+### 9. Taking into account the previous points, make both porting scripts for new backends and modifying already-deployed backend scripts as transparent as possible
 
 ## Realization of the goals
 
-Orleans framework does not have knowledge of deployment specific hardware (which may change during active deployment), the change of data during the deployment life-cycle and some vendor specific features are usable in only some situations. For this reason, the interface between relational database and Orleans should adhere a minimum set of abstractions and rules to meet the goals but to make it also robust against misuse and easy to test if needed.
-Runtime Tables, Cluster Management and the concrete [membership protocol implementation](https://github.com/dotnet/orleans/blob/master/src/Orleans/SystemTargetInterfaces/IMembershipTable.cs). Also, the SQL Server implementation contains SQL Server edition specific tuning.
+The Orleans framework does not have a knowledge of deployment-specific hardware (which hardware may change during active deployment), the change of data during the deployment life-cycle, and certain vendor-specific features which are only usable in certain situations. For this reason, the interface between the database and Orleans should adhere to a minimum set of abstractions and rules to meet these goals, make it robust against misuse, and make it easy to test if needed.
+Runtime Tables, Cluster Management and the concrete [membership protocol implementation](https://github.com/dotnet/orleans/blob/master/src/Orleans/SystemTargetInterfaces/IMembershipTable.cs). Also, the SQL Server implementation contains SQL Server edition-specific tuning.
```

Comment:
We can delete this partial sentence

Reply:
Some have late night sentenced there. I think anything making it clearer and better organized is an improvement. :)

The idea is basically that Orleans uses ADO.NET like this:
- Load a scripts with known identifiers.
- Prepare a call by giving ADO.NET the necessary parameters with their names and correct types.
- Call the database with the loaded and given parameters.

One analogue is that Orleans loads a JavaScript function and sends it over the wire with parameters. It's then the script that does whatever it does on the database to make operation efficient. The operator can for instance make use of compiled procedures and in-memory tables that are not universally available and have some slight limitations. Or the operator can maybe divert some types of stored elements to alternative storage managed via SQL Server.

This arrangement makes it possible to use engines that do not have a connector but understand the needed subset of ADO.NET compatible protocol (usually a dialect of PostgreSQL or MySQL one).

Reply:
@veikkoeeva, do you want me to incorporate what you have just explained?

Reply:
@voipengineer If you think it makes understanding the content easier. I don't know if it does. It's more like explaining more to you so it maybe gives more tools to understand and edit. You're doing a good job already.

Reply:
I think I'm going to quit while I'm ahead. If it's OK with you guys let's commit what's here.